python/google/protobuf/internal/type_checkers.py

```diff
@@ -57,7 +61,10 @@
 
 
 def TruncateToFourByteFloat(original):
-  return ctypes.c_float(original).value
+  if ctypes:
+    return ctypes.c_float(original).value
+  else:
+    return struct.unpack('<f', struct.pack('<f', original))[0]
```

Comment:
If this is correct, we might as well use `struct` all of the time.

Otherwise we won't get any CI coverage for this code path.

Reply:
From my understanding they are the same:

```
Python 3.10.9 (main, Dec 15 2022, 10:44:50) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import ctypes, struct
>>> original = 0.12345678901234567890123456789
>>> ctypes.c_float(original).value
0.12345679104328156
>>> struct.unpack('<f', struct.pack('<f', original))[0]
0.12345679104328156
```

I can switch this to always use struct. When putting up the PR I reverted this to what it was in the proto project in the past:

https://github.com/protocolbuffers/protobuf/blob/10d0faef6de038960ec825c003c85581800fba1c/python/google/protobuf/internal/type_checkers.py#L68-L72

It was changed to ctypes only in https://github.com/protocolbuffers/protobuf/commit/75eff10d81233d3d08ce5ded215e526c1fcace88 so I figured there may have been some reasoning as to why, but that's unfortunately not visible to OSS due to the big sync PR.

Reply:
Done.

Reply:
> but that's unfortunately not visible to OSS due to the big sync PR.

I looked back and it was part of a CL to remove Python 2.x compatibility code.

If `ctypes` is available, then the `ctypes` code is simpler. But if it's not always available, we might as well use `struct` all of the time.